### DISCUSSION TITLE
Exclude auto-generated response model title from doc

### DISCUSSION THREAD

**Discussion by chekhovana** [fastapi/fastapi#11448] [Questions] [answered]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None
    tags: list[str] = []


@app.get("/items/")
async def read_items() -> list[Item]:
    return [
        Item(name="Portal Gun", price=42.0),
        Item(name="Plumbus", price=32.0),
    ]
```

### Description

When method returns array of objects, corresponding response schema is documented with auto-generated title:
<img width="680" alt="Screenshot 2024-04-17 at 06 53 57" src="https://github.com/tiangolo/fastapi/assets/6779698/b949dbbd-f40b-48c1-b2ce-d21edf4e61db">

I suppose it's assigned here:
https://github.com/tiangolo/fastapi/blob/3425c834cc113060d00b8a295d3456eb306a109d/fastapi/_compat.py#L197C1-L203C1

Is there any way to skip this assignment?

### Operating System

macOS

### FastAPI Version

0.110.1

### Pydantic Version

2.7.0

### Python Version

3.9.6

**Comment by YuriiMotov** [comment]
Hi!
What do you want to see there instead of this long string?

You can create new type using RootModel and use it as a response model. In this case you will see the name of new type there:
```
from fastapi import FastAPI
from pydantic import BaseModel, RootModel

app = FastAPI()

class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None
    tags: list[str] = []

class ListOfItem(RootModel):
    root: list[Item]

@app.get("/items/")
async def read_items() -> ListOfItem:
    return [
        Item(name="Portal Gun", price=42.0),
        Item(name="Plumbus", price=32.0),
    ]
```

**Reply by chekhovana** [reply]
Hello, thanks for your reply!

It's a bit awkward to declare special type for the list of each object type.

I'd prefer to see the list of items, in the following format: `[Item]`
This is achieavable by manually commenting out string with `title` in yaml doc file:
<img width="1153" alt="Screenshot 2024-04-17 at 11 29 54" src="https://github.com/tiangolo/fastapi/assets/6779698/7cf3bb80-a846-4dfb-929e-e69cd63a2283">

But I can't get how to remove `title` programmatically.

**Reply by YuriiMotov** [reply, marked as answer]
You can customize OpenApi schema by using this approach:
```
from fastapi import FastAPI
from fastapi.openapi.utils import get_openapi
from pydantic import BaseModel

app = FastAPI()


class Item(BaseModel):
    name: str
    description: str | None = None
    price: float
    tax: float | None = None
    tags: list[str] = []



@app.get("/items/")
async def read_items() -> list[Item]:
    return [
        Item(name="Portal Gun", price=42.0),
        Item(name="Plumbus", price=32.0),
    ]


def custom_openapi():
    if app.openapi_schema:
        return app.openapi_schema
    openapi_schema = get_openapi(
        title=app.title,
        version=app.version,
        summary=app.summary,
        description=app.description,
        routes=app.routes,
    )

    for path in openapi_schema["paths"].values():
        for operation in path.values():
            try:
                response_200 = operation["responses"]["200"]
                schema = response_200["content"]["application/json"]["schema"]
                schema.pop("title")
            except KeyError:
                continue


    app.openapi_schema = openapi_schema
    return app.openapi_schema


app.openapi = custom_openapi
```

This will remove the `title` field from all response schemas, corresponding to status code 200 and content type `application/json`

**Reply by chekhovana** [reply]
thanks a lot, I’ll try
